specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2022-05-31/digitaltwins.json

```diff
@@ -0,0 +1,1622 @@
+        "description": "Retrieves a digital twin.\nStatus codes:\n* 200 OK\n* 400 Bad Request\n  * InvalidArgument - The digital twin id is invalid.\n* 404 Not Found\n  * DigitalTwinNotFound - The digital twin was not found.",
+        "x-ms-examples": {
+          "Get a digital twin by id": {
+            "$ref": "./examples/GetTwin.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/digitalTwinId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "object"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Default response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Twins"
+        ],
+        "operationId": "DigitalTwins_Add",
+        "description": "Adds or replaces a digital twin.\nStatus codes:\n* 200 OK\n* 400 Bad Request\n  * InvalidArgument - The digital twin id or payload is invalid.\n  * ModelDecommissioned - The model for the digital twin is decommissioned.\n  * TwinLimitReached - The maximum number of digital twins allowed has been reached.\n  * ValidationFailed - The digital twin payload is not valid.\n* 412 Precondition Failed\n  * PreconditionFailed - The precondition check (If-Match or If-None-Match) failed.",
+        "x-ms-examples": {
+          "Add a digital twin": {
+            "$ref": "./examples/PutTwinBasicExample.json"
+          },
+          "Add a digital twin (with properties and components)": {
+            "$ref": "./examples/PutTwinAdvancedExample.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/digitalTwinId"
+          },
+          {
+            "name": "twin",
+            "description": "The digital twin instance being added. If provided, the $dtId property is ignored.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "type": "object"
+            }
+          },
+          {
+            "$ref": "#/parameters/if-none-match-star"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type": "object"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "Asynchronous Success"
+          },
```

Comment:
these appear to be long-running operation

Reply:
We are currently not returning a 202 with this API - I believe a previous review of an upcoming preview API version recommended removing this response code from the swagger. Would that be considered a breaking change? (We don't actually ever return it).

Reply:
Actually, per a previous review (https://github.com/Azure/azure-rest-api-specs-pr/issues/5257), I think it's OK to remove.

Reply:
I've removed them. This was discussed in a previous API review (of an upcoming preview API).